doc/ensemble.rst

```diff
@@ -0,0 +1,739 @@
+********
+Ensemble
+********
+
+========
+Overview
+========
+An ``Ensemble`` is a SmartSim object comprised of multiple
+:ref:`Model<model_api>` objects, each representing an individual
+simulation. An Ensemble can be be reference as a single instance and
+enables users to launch a group of simulations
+at runtime. Ensembles can be launched with other SmartSim
+entities and ML infrastructure to build AI-enabled workflows.
+The :ref:`Ensemble API<ensemble_api>` offers key Ensemble features,
+including helper functions to:
+
+- :ref:`Ensemble.add_ml_model()<load_ml_model>`: load a TF, TF-lite, PT, or ONNX model into the database at runtime.
+- :ref:`Ensemble.add_function()<load_tf_function>`: load TorchScript functions to launch with each entity belonging to the ensemble.
+- :ref:`Ensemble.add_script()<load_script>`: load TorchScripts to launch with each entity at runtime.
+- :ref:`Ensemble.enable_key_prefixing()<prefix_ensemble>`: prevent key overlapping and enable ensemble members to run the same code.
+- :ref:`Ensemble.attach_generator_files()<attach_file>`: attach configuration files to assign ensemble member parameters or load parameters to file.
+
+An Ensemble is created through the ``Experiment.create_ensemble()`` function.
+There are **three** Ensemble configuration that users can adjust to customize
+Ensemble behavior:
+
+1. :ref:`Manually<append_init>`: Create each Model individually and subsequently append the simulation to the ensemble.
+2. :ref:`Replica creation<replicas_init>`: Generate a specified number of copies or instances of a simulation.
+3. :ref:`Parameter expansion<param_expansion_init>`: Explore a range of simulation variable assignments by
+   specifying values of input parameters and selecting an expansion strategy to rearrange
+   across multiple Models within the ensemble.
+
+==============
+Initialization
+==============
+--------
+Overview
+--------
+The creation of an Ensemble involves the use of various initializer arguments,
```

Comment:
1st sentence from L39-40 is sort of "intro to how functions work" and feels unnecessary.